dotrix_voxel/src/sdf/circle_trace.rs

```diff
@@ -0,0 +1,150 @@
+use crate::Grid;
+use crate::TexSdf;
+use dotrix_core::{
+    assets::{Mesh, Shader},
+    ecs::{Const, Mut, System},
+    renderer::{BindGroup, Binding, PipelineLayout, RenderOptions, Sampler, Stage},
+    Application, Assets, Globals, Renderer, World,
+};
+use dotrix_math::*;
+use dotrix_primitives::Cube;
+
+mod camera;
+mod lights;
```

Comment:
[question]
What is the reason of having these modules duplicated here, instead of using ones from the core?

Reply:
Ah yes, that's on my todo list. I lot of this code comes from my other project which deals with pure sdf functions (perfect spheres look so nice). In that code I do lights using storage buffers so that I can have more than 10. Rather than rework my code to use core I just copy pasted the whole lot as a set.

I'll work on it when I can but I'm quite busy with work and my toddler at the moment.

Reply:
The PR #158 has some of the improvements to the lighting that I would want merged before I try tackling using I have using lights from `dotrix_pbr`.